src/utils.py

```diff
@@ -119,12 +119,16 @@ def get_next_month(self):
         )
 
     def __eq__(self, other: "ParsedDataPoint"):
-        return self.month == other.month and self.year == other.year
+        return other is not None and self.month == other.month and self.year == other.year
```

Comment:
fungerer `return other and self.month == other.month and self.year == other.year`?